bin/agitator

```diff
@@ -40,57 +115,85 @@ function start_agitator() {
 
   if ${AGTR_HDFS:-false} ; then
     agitator_log=${log_base}_hdfs-agitator
-    sudo -u "$AGTR_HDFS_SUPERUSER" nohup "${libexec}/hdfs-agitator.pl" --sleep "${AGTR_HDFS_SLEEP_TIME}" --hdfs-cmd "${AGTR_HDFS_COMMAND}" --superuser "${AGTR_HDFS_SUPERUSER}" >"${agitator_log}.out" 2>"${agitator_log}.err" &
+    sudo -u "$AGTR_HDFS_SUPERUSER" nohup "${at_home}/libexec/hdfs-agitator.pl" --sleep "${AGTR_HDFS_SLEEP_TIME}" --hdfs-cmd "${AGTR_HDFS_COMMAND}" --superuser "${AGTR_HDFS_SUPERUSER}" >"${agitator_log}.out" 2>"${agitator_log}.err" &
   fi
 }
 
 function stop_agitator() {
   [[ -n $AGITATOR_USER ]] || AGITATOR_USER=$(whoami)
-  if [[ $AGITATOR_USER == "$AGTR_ACCUMULO_USER" ]];  then
-    echo "Stopping all processes matching 'datanode-agitator.pl' as $AGTR_HDFS_USER"
-    sudo -u "$AGTR_HDFS_USER" pkill -f datanode-agitator.pl 2>/dev/null
-    echo "Stopping all processes matching 'hdfs-agitator.pl' as $AGTR_HDFS_USER"
-    sudo -u "$AGTR_HDFS_USER" pkill -f hdfs-agitator.pl 2>/dev/null
-    echo "Stopping all processes matching 'agitator.pl' as $AGITATOR_USER"
-    pkill -f agitator.pl 2>/dev/null 2>/dev/null
-  else
-    echo "Stopping all processes matching 'datanode-agitator.pl' as $AGTR_HDFS_USER"
-    sudo -u "$AGTR_HDFS_USER" pkill -f datanode-agitator.pl 2>/dev/null
-    echo "Stopping all processes matching 'hdfs-agitator.pl' as $AGTR_HDFS_USER"
-    sudo -u "$AGTR_HDFS_USER" pkill -f hdfs-agitator.pl 2>/dev/null
-    echo "Stopping all processes matching 'agitator.pl' as $AGTR_ACCUMULO_USER"
-    sudo -u "$AGTR_ACCUMULO_USER" pkill -f agitator.pl 2>/dev/null
+  echo "Stopping all processes matching 'sleep' as $AGITATOR_USER"
+  pkill -f sleep 2>/dev/null
+  echo "Stopping all processes matching 'agitator' as $AGITATOR_USER"
+  pkill -f agitator 2>/dev/null
+}
+
+function parse_fail() {
+  echo "Failed to parse ${conf}/cluster.yaml"
+  exit 1
+}
+
+# Read the configuration from cluster.yaml - expects at least the manager and tservers
+function read_cluster_conf() {
+  conf="${ACCUMULO_HOME}/conf"
+  echo "Reading cluster config from ${conf}/cluster.yaml"
+  trap 'rm -f "$CONFIG_FILE"' EXIT
+  CONFIG_FILE=$(mktemp) || exit 1
+  accumulo org.apache.accumulo.core.conf.cluster.ClusterConfigParser "${conf}"/cluster.yaml > "$CONFIG_FILE" || parse_fail
+  . "$CONFIG_FILE"
+  rm -f "$CONFIG_FILE"
+
+  if [[ -z $MANAGER_HOSTS ]]; then
+    echo "ERROR: managers not found in ${conf}/cluster.yaml"
+    exit 1
+  fi
+  if [[ -z $TSERVER_HOSTS ]]; then
+    echo "ERROR: tservers not found in ${conf}/cluster.yaml"
+    exit 1
+  fi
+}
+
+# Given an app_name $1, set the array of hosts to the named reference passed in $2
+function get_app_hosts() {
+  local app_name=$1
+  local -n app_hosts=$2
+  if [[ "$app_name" == "manager" ]]; then
+    app_hosts=( $MANAGER_HOSTS )
+  fi
+
+  if [[ "$app_name" == "tserver" || "$app_name" == "datanode" ]]; then
+    app_hosts=( $TSERVER_HOSTS )
   fi
```

Comment:
The corresponding change here to my suggestion above could be:

```suggestion
  case "$1" in
    manager)  echo -n "$MANAGER_HOSTS" ;;
    tserver|datanode) echo -n "$TSERVER_HOSTS" ;;
    *) return 1 ;;
  esac
```

The `-n` is so you don't capture a newline as part of the last host when `mapfile` reads it using a space delimiter.